### DISCUSSION TITLE
Is there any compass support in the lates Corodova versions ?

### DISCUSSION THREAD

**Discussion by ghevge** [apache/cordova#410] [Q&A]
Is there compass support in the lates corodova versions ?
I'm more interested in ios support. For Android I don't really need it as I can use the webview support.
Going over the Cordova documentation, I can see in some older versions, references to compass functionality. But those references were removed in later versions.

I've also found this plugin: cordova-plugin-device-orientation but it seems unsupported as there were no new releases since 2018.

Thanks

**Comment by BigBalli** [comment]
Have you tried `window.addEventListener("deviceorientation", compassHandler);` or `navigator.compass.watchHeading();` ?

**Comment by breautek** [comment]
> I've also found this plugin: cordova-plugin-device-orientation but it seems unsupported as there were no new releases since 2018.

not having a new release isn't a strong indicator that it isn't maintained. Afaik have been no changes in the native SDK that required us to make a change in the plugin.

If I recall correctly the plugint was once deprecated due to it being supported in the native webview, but in iOS 13+ they started adding permission modals and other privacy features that made using a native implementation more desirable, so it was brought back.

**Comment by ghevge** [comment]
deviceorientation works on iso. Thanks!
